interface/src/lib.rs

```diff
@@ -5,16 +5,67 @@
 mod context;
 pub use context::*;
 
-use blake3::Hasher;
+use blake3::{hash, Hasher};
+use serde::{Deserialize, Serialize};
 
-use std::{collections::BTreeMap, convert::TryFrom, error::Error, fmt};
+use std::{cmp::Ordering, collections::BTreeMap, convert::TryFrom, error::Error, fmt};
 
 pub const ADDRESS_LENGTH: usize = 24;
 pub const BYTES32_LENGTH: usize = 32;
+pub const METHOD_SELECTOR_LENGTH: usize = 4;
 pub type Address = [u8; ADDRESS_LENGTH];
 pub type Balance = u64;
 pub type Bytes32 = [u8; BYTES32_LENGTH];
 pub type Bytes = [u8];
+pub type MethodSelectorBytes = [u8; METHOD_SELECTOR_LENGTH];
+pub const METHOD_SELECTOR_DEFAULT: MethodSelector = MethodSelector {
+  bytes: [0u8; METHOD_SELECTOR_LENGTH],
+};
+
+#[derive(Debug, Clone, Deserialize, Eq, PartialEq, PartialOrd, Serialize)]
+pub struct MethodSelector {
+  bytes: MethodSelectorBytes,
+}
+
+impl MethodSelector {
+  pub fn bytes(&self) -> MethodSelectorBytes {
+    self.bytes
+  }
+}
+
+impl From<&str> for MethodSelector {
+  fn from(value: &str) -> Self {
+    let h = hash(value.as_bytes());
+    let vbytes = h.as_bytes();
+    MethodSelector {
+      bytes: vbytes[vbytes.len() - METHOD_SELECTOR_LENGTH..]
+        .try_into()
+        .unwrap(),
+    }
+  }
```

Comment:
NIT: it would be simpler to take the first bytes, wdyt?
```suggestion
    MethodSelector(..METHOD_SELECTOR_LENGTH].try_into().unwrap())
  }
```

Reply:
yes, you're right that it's simpler and it makes no difference